### PR TITLE
Simple EDS API Aligned with commonalities

### PR DESCRIPTION
**Pull request by Kevsy**
* Unused parameters removed
* Unused resource types removed
* `filter` parameter added (to future-proof  `GET /mecplatforms` with no filter as a request to read the collection resource)
* URI changed to kebab-case
* OperationId updated to Camel Case
*  Identifiers moved to header for security reasons and aligned withe identifier names in  Device Identifier API and QoD API
* Basepath updated including major version
* Standard error codes added
* Documentation simplified and examples added
* following review: success response made inline

### CODE REVIEW

**Review by kevin8xu — APPROVED** [review]
Looks good to me.

File: code/API_definitions/Discovery/simple_edge_discovery.yaml [hunk 1]
```diff
@@ -1,392 +1,427 @@
...
+          in: query
+          required: true
+          description: filter the MEC Platforms according to the parameter value. For this API the only supported value is `closest`
+          schema:
+            type: string
+            enum: [
+              "closest"
+              ]
+            
+        - name: IP-Address
```
[the remark this answers is not in the document]
Comment (Kevsy) [reply code/API_definitions/Discovery/simple_edge_discovery.yaml:126]: Thanks for reviewing @RandyLevensalor -

**Comment by Kevsy** [comment]
Following inital reviews (thanks!)  I've made thgree changes to the commit (same Pull Request):

1. made `MecPlatforms` an array of `MecPlatform`. Only one `MecPlatform` is returned for this Simple EDS API but this allows extensibility
2. Added the Provider as context to the Ern (name of closest MEC Platform) as there is no agreed globally unique identifier for a MEC Platform.
3. Simplified the introduction.

File: code/API_definitions/Discovery/simple_edge_discovery.yaml [hunk 2]
```diff
@@ -1,392 +1,444 @@
...
+    
+    ## Success
+    
+    A JSON object is returned containing a `MECPlatforms` array with a single member, along with the HTTP `200 OK` status code. The value of the Ern (Edge Resource Name) object is the name of the closest MEC platform to the user device. An example of this JSON object is as follows:
+    ```
+    {
+      "MecPlatforms": [
+        {
+          "Provider": "AWS",
+          "Ern": "eu-west-2-wl1-man-wlz-1
```
Comment (RandyLevensalor) [inline code/API_definitions/Discovery/simple_edge_discovery.yaml:72]: Missing closing " at the end of eu-west-2-wl1-man-wlz-1

**Comment by gunjald** [comment]
Can we assume that there is an implicit assumption that the MEC platform names are to be unique within an telco environment? As the API client responsibility is assumed to manage the mapping a MEC platform name as returned in the API response to the application  endpoints or IP address details so in my opinion it implicitly imposes a uniqueness constraints for the operator to manage the uniqueness of MEC platform names. 

If this is the case then it should be mentioned in the API description as the constraint then needs to be considered for SED API implementation.

File: code/API_definitions/Discovery/simple_edge_discovery.yaml (hunk 2 above)
Comment (Kevsy) [reply code/API_definitions/Discovery/simple_edge_discovery.yaml]: Thanks @RandyLevensalor , fixed with latest commit

**Comment by Kevsy** [comment]
> Can we assume that there is an implicit assumption that the MEC platform names are to be unique within an telco environment? As the API client responsibility is assumed to manage the mapping a MEC platform name as returned in the API response to the application endpoints or IP address details so in my opinion it implicitly imposes a uniqueness constraints for the operator to manage the uniqueness of MEC platform names.
> 
> If this is the case then it should be mentioned in the API description as the constraint then needs to be considered for SED API implementation.

Added to latest document a 'note for API publishers' that the `provider` : `ern` tuple must be unique. The cloud provider zone names are already unique within the cloud provider domain, so it's really a comment to operators with self-build MEC. I would already strongly suspect those to be unique within a given operator, but best to make it explicit :)

File: code/API_definitions/Discovery/simple_edge_discovery.yaml [hunk 3]
```diff
@@ -1,392 +1,446 @@
...
+        clientCredentials:
+          tokenUrl: "{tokenUrl}"
+          scopes: {}
   schemas:
-    Resources:
+    successResponse:
+      type: object
+      properties:
+        MecPlatforms:
+          $ref: '#/components/schemas/MecPlatforms'
```
[the remark this answers is not in the document]
Comment (Kevsy) [reply code/API_definitions/Discovery/simple_edge_discovery.yaml]: Thanks @gainsley, that's a good suggestion: updated accordingly in [latest commit](https://github.com/camaraproject/EdgeCloud/pull/127/commits/1505ff7e694f3810ebdf1194f927bba5d8a6c471)

**Review by sergiofranciscoortiz — APPROVED** [review]
Approved after last changes taking into account previous comments.

**Review by RandyLevensalor — APPROVED** [review]

**Review by gunjald — APPROVED** [review]
Reviewed

**Pull request merged**